pkg/alertmanager/operator.go

```diff
@@ -799,7 +799,7 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 
 	if ok && sErr.ErrStatus.Code == 422 && sErr.ErrStatus.Reason == metav1.StatusReasonInvalid {
 		c.metrics.StsDeleteCreateCounter().Inc()
-		level.Info(c.logger).Log("msg", "resolving illegal update of Alertmanager StatefulSet", "details", sErr.ErrStatus.Details)
+		level.Info(c.logger).Log("msg", "recreating AlertManager StatefulSet because the update operation wasn't possible", "reason", sErr.ErrStatus.Details.Causes[0].Message)
```

Comment:
Why listing only first error reason (`ErrStatus.Details.Causes[0].Message`)?

Reply:
agreed we could join all messages (though in practice there's only one message).

Reply:
sErr.ErrStatus.Details this is how details looked in the output, I could see one message only

```
details="&StatusDetails{Name:alertmanager-main,Group:apps,Kind:StatefulSet,Causes:[]StatusCause{StatusCause{Type:FieldValueForbidden,Message:Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden,Field:spec,},},RetryAfterSeconds:0,UID:,}
```

Let me see how to join messages